pkg/storage/storage_provisioning.go

```diff
@@ -62,24 +67,57 @@ func NewProvisionImpl(c client.Client) *ProvisionImpl {
 	}
 }
 
-func equal(required *ProvisionedBucket, existing *comv1alpha1.DatasetSpec) bool {
-	if required.Name != existing.Local["bucket"] {
+func newDatasetAsUnstructured(name string, namespace string) *unstructured.Unstructured {
+	object := &unstructured.Unstructured{}
+	object.SetGroupVersionKind(schema.GroupVersionKind{Group: GroupVersion.Group, Version: GroupVersion.Version, Kind: "Dataset"})
+	object.SetNamespace(namespace)
+	object.SetName(name)
+	return object
+}
+
+func (r *ProvisionImpl) getDatasetAsUnstructured(name string, namespace string) (*unstructured.Unstructured, error) {
+	object := &unstructured.Unstructured{}
+	object.SetGroupVersionKind(schema.GroupVersionKind{Group: GroupVersion.Group, Version: GroupVersion.Version, Kind: "Dataset"})
+	object.SetNamespace(namespace)
+	object.SetName(name)
+
+	objectKey := client.ObjectKeyFromObject(object)
+
+	if err := r.Client.Get(context.Background(), objectKey, object); err != nil {
+		return nil, err
+	}
+	return object, nil
+}
+
+func getValue(obj map[string]interface{}, path []string) string {
```

Comment:
Consider rewriting as a variadic function (path ...string) to simplify calling code.

Reply:
OK.